src/lib/components/dateSelector/index.tsx

```diff
@@ -111,6 +113,22 @@ const DateSelector = ({
     .endOf('year')
     .toDate();
 
+  useEffect(() => {
+    const handleOutsideClick = (event: MouseEvent) => {
+      if (!calendarContainerRef || !calendarContainerRef.current) {
+        return;
+      }
+      if (!calendarContainerRef.current.contains(event.target as Node)) {
+        setOpenCalendar(false);
+      }
+    };
+    document.addEventListener('mousedown', handleOutsideClick);
+
+    return () => {
+      document.removeEventListener('mousedown', handleOutsideClick);
+    };
+  }, [calendarContainerRef, setOpenCalendar]);
+
```

Comment:
WDYT about moving this to a the utils or hooks folder? Seems like something that might be useful for other cases. We could even export it and take advantage of that in app/website/others repos

Reply:
Was going to make the same comment!

Reply:
Makes sense.

Reply:
We could even move that hook to dirty-swan and import it on website, less duplicated code 🧹 🧹

Reply:
Thanks @ruizhen88 @diogomateus for the feedback. Will check and then update.

Reply:
Ok converted the onClickOutside into a hook @diogomateus @ruizhen88

Reply:
Looks good! Should we add some tests as well?

Reply:
Ok, added some test @diogomateus